src/webgpu/util/floating_point.ts

```diff
@@ -265,6 +265,16 @@ export type FPMatrix =
       [FPInterval, FPInterval, FPInterval, FPInterval]
     ];
 
+export type FPValue = FPInterval | FPVector | FPMatrix | FPStruct;
+
+/** Container representing a struct of FP values */
+export class FPStruct {
```

Comment:
The choice of `FPStruct` was a little surprising.
In WGSL, a struct doesn't need to have members of the same type, and `frexp` returns a structure of `f32` and `i32`, which presumably this can't be used for.

Its not clear what the rules are for this. Do each of the members need to be the same FP type, or can you mix a f32 with a f16?